### PR TITLE
[doc / test / mempool] unbroadcast follow-ups

### PR DESCRIPTION
**Pull request by amitiuttarwar**
This PR is a follow up to #18038 which introduced the idea of an unbroadcast set & focuses mostly on documentation updates and test fixes. One small functionality update to not throw an expected error in `LoadMempool` when you upgrade software versions.

#18895 is another follow up to that addresses other functionality updates.

Background context: 
The unbroadcast set is a mechanism for the mempool to track locally submitted transactions (via wallet or RPC). The node does a best-effort of delivering the transactions to the network via retries every 10-15 minutes until either a `GETDATA` is received or the transaction is removed from the mempool.

### CODE REVIEW

File: doc/release-notes.md [hunk 1]
```diff
@@ -87,6 +92,13 @@ New settings
 Wallet
 ------
 
+- To improve wallet privacy, the frequency of wallet rebroadcast attempts has
+  been reduced from approximately once every 15 minutes to approximately once a
+  day. To maintain a similar level of guarantee for initial broadcast of wallet
```
Comment (jonatack) [inline doc/release-notes.md:97]: suggest s/approximately once a day/once every 12-36 hours/

File: test/functional/wallet_resendwallettransactions.py [hunk 2]
```diff
@@ -69,9 +54,17 @@ def run_test(self):
...
+        # rebroadcast
+        rebroadcast_time = int(time.time()) + 12 * 60 * 60
+        node.setmocktime(rebroadcast_time)
+        node.p2ps[1].sync_with_ping()
+        assert_equal(node.p2ps[1].tx_invs_received[txid], 0)
+
+        self.log.info("Bump time & check that transaction is rebroadcast")
+        # Transaction should be rebroadcast approximately 24 hours in the future,
+        # but can range from 12-36. So bump 36 hours to be sure.
+        rebroadcast_time = int(time.time()) + 36 * 60 * 60
```
Comment (jonatack) [inline test/functional/wallet_resendwallettransactions.py:67]: Suggestion as per https://github.com/bitcoin/bitcoin/pull/18038#discussion_r415750840 with only one call to `time.time()` and two info logging statements.

```python
        # Check transaction is rebroadcast in 12-36 hour time interval.
        time_now = int(time.time())

        self.log.info("Bump time 12 hours from now, check transaction is not yet rebroadcast")
        rebroadcast_time = time_now  + (12 * 60 * 60)
        node.setmocktime(rebroadcast_time)
        assert_equal(node.p2ps[1].tx_invs_received[txid], 0)

        self.log.info("Bump time 36 hours from now, check transaction is rebroadcast")
        rebroadcast_time = time_now + (36 * 60 * 60)
        node.setmocktime(rebroadcast_time)
        wait_until(lambda: node.p2ps[1].tx_invs_received[txid] >= 1, lock=mininode_lock)
```

**Review by jonatack — COMMENTED** [review]
Quick first pass, will review after rebase.

File: src/net_processing.cpp [hunk 3]
```diff
@@ -810,6 +810,20 @@ void PeerLogicValidation::InitializeNode(CNode *pnode) {
...
+void PeerLogicValidation::ReattemptInitialBroadcast(CScheduler& scheduler) const
+{
+    std::set<uint256> unbroadcast_txids = m_mempool.GetUnbroadcastTxs();
+
+    for (const uint256& txid : unbroadcast_txids) {
+        RelayTransaction(txid, *connman);
+    }
+
+    // schedule next run for 10-15 minutes in the future
+    // add randomness on every cycle to avoid possibility of p2p fingerprinting
```
Comment (jnewbery) [inline src/net_processing.cpp:822]: Format in sentences please

**Review by jnewbery — COMMENTED** [review]
Concept ACK. Will review fully once this is rebased.

**Comment by fjahr** [comment]
Concept ACK

**Comment by brakmic** [comment]
Concept ACK.

Built, run and tested the code on macOS Catalina 10.15.4. Both tests were successful.
Commit 0dd03ca9b3531ac8c4ede030bb075b882b588f79

```shell
./test/functional/mempool_unbroadcast.py   (13s 854ms) │
2020-04-30T16:35:16.018000Z TestFramework (INFO): Initializing test directory /var/folders/7q/4ffytzk562dd2ky4bfg9_w7h0000gn/T/bitcoin_func_test_ngy9nfi0
2020-04-30T16:35:19.438000Z TestFramework (INFO): Test that mempool reattempts delivery of locally submitted transaction
2020-04-30T16:35:19.994000Z TestFramework (INFO): Generate transactions that only node 0 knows about
2020-04-30T16:35:21.479000Z TestFramework (INFO): Reconnect nodes & check if they are sent to node 1
2020-04-30T16:35:25.579000Z TestFramework (INFO): Add another connection & ensure transactions aren't broadcast again
2020-04-30T16:35:27.753000Z TestFramework (INFO): Test that transactions removed from mempool are removed from unbroadcast set
2020-04-30T16:35:28.071000Z TestFramework (INFO): Stopping nodes
2020-04-30T16:35:28.595000Z TestFramework (INFO): Cleaning up /var/folders/7q/4ffytzk562dd2ky4bfg9_w7h0000gn/T/bitcoin_func_test_ngy9nfi0 on exit
2020-04-30T16:35:28.596000Z TestFramework (INFO): Tests successful
```

```shell
./test/functional/wallet_resendwallettransactions.py  (13s 961ms)  
2020-04-30T16:36:05.847000Z TestFramework (INFO): Initializing test directory /var/folders/7q/4ffytzk562dd2ky4bfg9_w7h0000gn/T/bitcoin_func_test_k0jcybzc
2020-04-30T16:36:07.329000Z TestFramework (INFO): Create a new transaction and wait until it's broadcast
2020-04-30T16:36:17.090000Z TestFramework (INFO): Create a block
2020-04-30T16:36:17.213000Z TestFramework (INFO): Bump time & check that transaction is rebroadcast
2020-04-30T16:36:17.584000Z TestFramework (INFO): Stopping nodes
2020-04-30T16:36:18.055000Z TestFramework (INFO): Cleaning up /var/folders/7q/4ffytzk562dd2ky4bfg9_w7h0000gn/T/bitcoin_func_test_k0jcybzc on exit
2020-04-30T16:36:18.055000Z TestFramework (INFO): Tests successful
```

Will do a full review after rebase.

**Comment by Sjors** [comment]
> the node will try to announce unbroadcast transactions until a peer requests it via a `GETDATA`

If I was a bully node, I would send a `GETDATA` back to any node who announces a transaction I haven't seen yet. Is that kind of behaviour punished at the p2p level? Would it make sense to require that the `GETDATA` comes from a node that we never told about the transaction? Or that it comes from at least two nodes?

**Comment by amitiuttarwar** [comment]
@gzhao408 has kindly taken over the follow-ups that change the codebase functionality in #18807.

I'll use this PR to focus on updates to tests & docs. but currently still WIP. 

-----

@Sjors 
> If I was a bully node, I would send a GETDATA back to any node who announces a transaction I haven't seen yet. Is that kind of behaviour punished at the p2p level?

nodes requesting announced txns they haven't seen is ... desired :) but I think what you're getting at is the attacker might request but then not propagate. I don't see a way a peer could directly know if the other node propagated the transaction. So yes, we'd have to use another heuristic to try to evaluate that question. 

but something to keep in mind.. transactions on the unbroadcast set are queued for every peer (`ReattemptInitialBroadcast` calls `RelayTransaction` which iterates through each node). so the attacker sending back a `GETDATA` would also have to somehow interrupt me from sending out to the other peers, be my only connection, etc. 

> Would it make sense to require that the GETDATA comes from a node that we never told about the transaction?

we don't usually get unsolicited `GETDATA` messages, but maybe what you're considering here is getting an `INV` from a peer we didn't announce it to. I considered this sort of approach in earlier iterations of the design (announce to some peers, wait for it to come back from others), but it seems unnecessary & a significant change to current txn propagation. 

> Or that it comes from at least two nodes?

also considered this approach, but over the course of review comments it didn't seem necessary bc some of the reasons talked about above. all of this design discussion is scattered throughout #18038 if you're interested in digging deeper :)

File: doc/release-notes.md (hunk 1 above)
Comment (amitiuttarwar) [reply doc/release-notes.md]: sure. changed

File: test/functional/wallet_resendwallettransactions.py (hunk 2 above)
Comment (amitiuttarwar) [reply test/functional/wallet_resendwallettransactions.py]: so, I took a closer look at this part of the test, and as @jnewbery pointed out [in that thread](https://github.com/bitcoin/bitcoin/pull/18038#discussion_r417408126), the test wasn't actually checking anything (not just the additional check after 12 hour bump, but the existing one as well).

I've updated so there's just one "transaction should not be rebroadcast" check, and that it occurs after we've bumped almost 12 hours into the future.

File: test/functional/wallet_resendwallettransactions.py [hunk 4]
```diff
@@ -49,16 +49,21 @@ def run_test(self):
         block.solve()
         node.submitblock(ToHex(block))
 
-        # Transaction should not be rebroadcast
         node.syncwithvalidationinterfacequeue()
-        node.p2ps[1].sync_with_ping()
+        now = int(time.time())
```
Comment (amitiuttarwar) [inline test/functional/wallet_resendwallettransactions.py:53]: @MarcoFalke I removed the `sync_with_ping()` call because I think its a no-op, but I'd like your review to make sure I'm not (re)introducing a race in weird conditions. 

my understanding: 
`sync_with_ping()` is a p2p interface method that ensures any messages it has sent to the node have been processed by the node. 

the `syncwithvalidationinterfacequeue()` makes sense to me bc that's making sure the wallet gets notified about the block.

since we're trying to check if the wallet kicks off rebroadcasting the transaction, so this synchronization should not be relevant? it was definitely not sufficient. the test previously wasn't checking anything bc it didn't give enough time for a rebroadcast to kickoff, so I've added a `time.sleep()` call. 

tldr; I removed the `sync_with_ping()` bc I think it was a no-op & misleading. I've added a `time.sleep(2)` to actually test that a transaction isn't rebroadcast. please lmk if I'm missing something.

File: src/net_processing.cpp (hunk 3 above)
Comment (amitiuttarwar) [reply src/net_processing.cpp]: fixed

**Comment by amitiuttarwar** [comment]
review comments addressed, additional release notes added & WIP removed.

File: test/functional/wallet_resendwallettransactions.py (hunk 4 above)
Comment (MarcoFalke) [reply test/functional/wallet_resendwallettransactions.py:53]: >  sync_with_ping() is a p2p interface method that ensures any messages it has sent to the node have been processed by the node.

Correct. Also, any messages that the node has put in the send buffer (before the pong is sent) will get "flushed out" to the mininode.

File: doc/release-notes.md [hunk 5]
```diff
@@ -65,6 +65,14 @@ Notable changes
 P2P and network changes
 -----------------------
 
+- The mempool now tracks transactions submitted via the wallet or rpcs as
+  unbroadcast transactions. Every 10-15 minutes, the node will try to announce
+  unbroadcast transactions until a peer requests it via a `GETDATA`, or the
```
Comment (MarcoFalke) [inline doc/release-notes.md:70]:
```suggestion
  unbroadcast transactions until a peer requests it via a `getdata` message, or the
```

File: doc/release-notes.md [hunk 6]
```diff
@@ -65,6 +65,14 @@ Notable changes
...
+  transaction is removed from the mempool for other reasons. If you have a
+  custom setup to use P2P messages to submit a local transaction to your node
```
Comment (MarcoFalke) [inline doc/release-notes.md:72]:
```suggestion
  custom setup to use P2P messages to submit a transaction that belongs to a loaded wallet to your node
```

local means wallet, right?

**Review by maflcko — COMMENTED** [review]
some questions on the release notes

File: test/functional/wallet_resendwallettransactions.py (hunk 4 above)
Comment (amitiuttarwar) [reply test/functional/wallet_resendwallettransactions.py:53]: 👍 . so, do you agree that seemed irrelevant here?

File: doc/release-notes.md (hunk 6 above)
Comment (amitiuttarwar) [reply doc/release-notes.md:72]: kind of.. I've been using the language "local" because it could be wallet, or submitted via `sendrawtransaction`. similarly, if we're talking about these edge cases, someone could theoretically form a transaction & give it to me out of band, then I submit from one computer to my main node via my highly specific custom P2P setup that involves being disconnected from the greater internet. but I guess every valid transaction involves some loaded wallet? 

lol. it all seems like quite an edge case. maybe here I can just remove the word "local".

new:
> If you have a custom setup to use P2P messages to submit a transaction to your node while it is not connected to other nodes, these changes reduce your initial broadcast guarantees. (#18038)

Comment (MarcoFalke) [reply doc/release-notes.md:72]: > If you have a custom setup to use P2P messages to submit a transaction to your node while it is not connected to other nodes, these changes reduce your initial broadcast guarantees

This has always been the case. Your pull didn't change that.

Comment (amitiuttarwar) [reply doc/release-notes.md:72]: oh. great point. lol ok I'll update to wallet.

Comment (amitiuttarwar) [reply doc/release-notes.md:72]: is there a reason you've suggested "that belongs to a loaded wallet" rather than just wallet transaction?

File: doc/release-notes.md (hunk 5 above)
Comment (amitiuttarwar) [reply doc/release-notes.md]: updated

File: doc/release-notes.md (hunk 6 above)
Comment (amitiuttarwar) [reply doc/release-notes.md]: updated to "wallet transaction", lmk if you have further thoughts

Comment (MarcoFalke) [reply doc/release-notes.md]: how would bitcoin core know that this is a wallet transaction when the wallet is not loaded? Maybe this is clear to everyone reading the sentence, but I think the distinction should be made to avoid potential confusion.

Comment (amitiuttarwar) [reply doc/release-notes.md]: ah, this was my misunderstanding. I was interpreting "loaded wallet" to mean <<wallet has spendable funds aka utxos>> vs <<node has wallet component loaded>>

also looked in the code and realized that watchonly is a wallet only thing. 

I get what you're saying now. I'm thinking about how to clarify the language.

File: test/functional/wallet_resendwallettransactions.py [hunk 7]
```diff
@@ -49,16 +49,21 @@ def run_test(self):
...
+
+        # Transaction should not be rebroadcast within first 12 hours
+        # Leave 2 mins for buffer
+        twelve_hrs = 12 * 60 * 60
+        two_min = 2 * 60
+        node.setmocktime(now + twelve_hrs - two_min)
+        time.sleep(2) # ensure enough time has passed for rebroadcast attempt to occur
         assert_equal(node.p2ps[1].tx_invs_received[txid], 0)
```
Comment (gzhao408) [inline test/functional/wallet_resendwallettransactions.py:61]: Would you want to grab the lock? And also since this function is pretty handy?
```suggestion
        assert_equal(txid in node.p2ps[1]get_invs(), False)
```

File: doc/release-notes.md [hunk 8]
```diff
@@ -65,6 +65,14 @@ Notable changes
...
+- The mempool now tracks transactions submitted via the wallet or rpcs as
+  unbroadcast transactions. Every 10-15 minutes, the node will try to announce
+  unbroadcast transactions until a peer requests it via a `getdata` message or
+  the transaction is removed from the mempool for other reasons. If you have a
+  custom setup to use P2P messages to submit a wallet transaction to your node
+  while it is not connected to other nodes, these changes reduce your initial
+  broadcast guarantees. (#18038)
+
 Updated RPCs
 ------------
```
Comment (gzhao408) [inline doc/release-notes.md:78]: Since it seems you'll need to include the rpc info [from #18895](https://github.com/bitcoin/bitcoin/pull/18895#issuecomment-630998986) I wanted to help draft a description:
```suggestion
- `getmempoolinfo` now returns an additional `unbroadcastcount` field which represents the number of transactions that the mempool considers unbroadcast.
- Mempool RPCs such as `getmempoolentry` and `getrawmempool` with `verbose=true`now return an additional "unbroadcast" field which indicates whether the mempool considers this transaction unbroadcast.
```

**Review by glozow — COMMENTED** [review]
Been trying to keep an eye on this PR since it's closely related to #18895 so I thought I'd leave a couple comments :)

**Comment by fanquake** [comment]
There's a comment from @harding [here](https://github.com/bitcoin/bitcoin/pull/18895#discussion_r429535717) about a change you might want to include in this PR.

File: test/functional/wallet_resendwallettransactions.py (hunk 7 above)
Comment (amitiuttarwar) [reply test/functional/wallet_resendwallettransactions.py]: good call, updated.

File: doc/release-notes.md (hunk 8 above)
Comment (amitiuttarwar) [reply doc/release-notes.md:78]: thanks! added with some extra info

File: doc/release-notes.md (hunk 6 above)
Comment (amitiuttarwar) [reply doc/release-notes.md]: ok. recruited harding & revamped the description. @MarcoFalke what do you think now? 

@harding thanks so much for your help. I continued tinkering with the language, let me know if you have any further recommendations

**Comment by amitiuttarwar** [comment]
rebased the branch, addressed review comments, added release notes for RPC updates from #18895 (thanks for getting that started @gzhao408) and updated language as per https://github.com/bitcoin/bitcoin/pull/18895#discussion_r429535717

File: doc/release-notes.md [hunk 9]
```diff
@@ -65,9 +65,28 @@ Notable changes
...
 
+- `getmempoolinfo` now returns an additional `unbroadcastcount` field. The
+  mempool tracks locally submitted transactions until their initial broadcast
+  is acknowledged by a peer. This field returns the count of transactions
+  waiting for acknowledgement.
+
+- Mempool RPCs such as `getmempoolentry` and `getrawmempool` with
+  `verbose=true` now return an additional "unbroadcast" field. This indicates
```
Comment (MarcoFalke) [inline doc/release-notes.md:85]:
```suggestion
  `verbose=true` now return an additional `unbroadcast` field. This indicates
```

**Review by maflcko — APPROVED** [review]
Looks good now. Thx

File: doc/release-notes.md [hunk 10]
```diff
@@ -65,9 +65,28 @@ Notable changes
 P2P and network changes
 -----------------------
 
+- The mempool now tracks whether transactions submitted via the wallet or rpcs
```
Comment (jonatack) [inline doc/release-notes.md:68]: suggestion: s/rpcs/RPCs/

File: doc/release-notes.md [hunk 11]
```diff
@@ -65,9 +65,28 @@ Notable changes
...
 
+- `getmempoolinfo` now returns an additional `unbroadcastcount` field. The
+  mempool tracks locally submitted transactions until their initial broadcast
+  is acknowledged by a peer. This field returns the count of transactions
+  waiting for acknowledgement.
+
+- Mempool RPCs such as `getmempoolentry` and `getrawmempool` with
+  `verbose=true` now return an additional "unbroadcast" field. This indicates
+  whether initial broadcast of the transaction has been acknowledged by a
+  peer. `getmempoolancestors` and `getmempooldescendants` are also impacted.
```
Comment (jonatack) [inline doc/release-notes.md:87]: I think the word "impacted" is odd here as it is often used to denote a negative connotation. Perhaps "updated", "also benefit from this change", etc.

File: doc/release-notes.md [hunk 12]
```diff
@@ -87,6 +106,13 @@ New settings
 Wallet
 ------
 
+- To improve wallet privacy, the frequency of wallet rebroadcast attempts has
+  been reduced from approximately once every 15 minutes to once every 12-36
```
Comment (jonatack) [inline doc/release-notes.md:110]: Each of these three release note entries employs a different verb tense: future ("will try/will not track") in the first entry, present ("now returns/tracks/returns") in the second one, and a form of past ("has been reduced") tense at the beginning of the third one. Perhaps use a consistent verb tense.

File: src/txmempool.h [hunk 13]
```diff
@@ -714,12 +714,12 @@ class CTxMemPool
...
 
     /** Returns transactions in unbroadcast set */
-    const std::set<uint256> GetUnbroadcastTxs() const {
+    std::set<uint256> GetUnbroadcastTxs() const {
         LOCK(cs);
         return m_unbroadcast_txids;
     }
 
-    // Returns if a txid is in the unbroadcast set
+    /** Returns if a txid is in the unbroadcast set */
```
Comment (jonatack) [inline src/txmempool.h:722]: s/if/whether/

(it does not return only if true)

**Review by jonatack — COMMENTED** [review]
Light review with a few minor suggestions below (feel free to ignore).

File: doc/release-notes.md (hunk 12 above)
Comment (amitiuttarwar) [reply doc/release-notes.md]: hm, I updated this one to be present ("has been reduced" -> "is reduced"). 

I read through the other ones and the difference between present & future makes sense to me... eg.
> The mempool now tracks whether transactions submitted via the wallet or RPCs
  have been successfully broadcast. Every 10-15 minutes, the node will try to
  announce unbroadcast transactions until...

the present "now tracks" indicates how the code has changed, but the future "will try" is what happens when you run the code in the future. 

so I've left them for now. if you have any specific suggestions, I'm happy to consider them.

**Comment by amitiuttarwar** [comment]
thanks for review @MarcoFalke & @jonatack, comments have been addressed !

**Review by glozow — COMMENTED** [review]
ACK [`9e1cb1a`](https://github.com/bitcoin/bitcoin/pull/18807/commits/9e1cb1adf1800efe429e348650931f2669b0d2c0)
And thank you for touching things up for me :)

**Comment by MarcoFalke** [comment]
ACK 9e1cb1adf1 👁

<details><summary>Show signature and timestamp</summary>

Signature:

```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

ACK 9e1cb1adf1 👁
-----BEGIN PGP SIGNATURE-----

iQGzBAEBCgAdFiEE+rVPoUahrI9sLGYTzit1aX5ppUgFAlwqrYAACgkQzit1aX5p
pUg2Dwv9EQ7oHPs8SFa494QyeEFubUt6HHSV8eJl98uS1jhK6ty7WCcKvHToqZMe
PiC8Jrwjk53dWMNcFMFvSbhU2x8I5qIQjLKj/4YNLOwZaUKasOiAwGwxoNnWkXNU
MT5odexk+BnkuOtYZN24qteCcZdCGD4dQqT//UC4CmG9aR0vi2Ff+y45Qn5FOfrB
NKR8OrgWh/AjTxrXCJMK73lGEscLuf0nmgOAqBJxt+VsONPpO2642RdAHASSqgD6
mXBIIbZ9dEOOCVQorwuKu9lyLvZ1LRcz3hcZfTQLiVA1bQBnqEyyVDNv/rSl7JVS
0O+dFSXu+YkgJMmVkFyhRq3/Le69bbK4y44am2w6Ab4W+hhzLnR1LGJGf5iL53Zc
QVQpqAEp7Zri2Vm5zZfqFK4y9sxqdmE8DZ4HrtljPT30Li8Hc7Ygo7cP0kVHPBhK
MU3sT5r8T/+V8p3VOj6L1XVgHPH3D97xyg4kPaDFeUoK0UysKfFfPVME50ZXQ27p
SNm8OtZI
=3KJW
-----END PGP SIGNATURE-----
```

[Timestamp of file with hash `de6ba48e94c3bcb5032ea5569ce708038f02ef67193d1448110f51b57e5d3132  -`](https://opentimestamps.org/info/?004f70656e54696d657374616d7073000050726f6f6600bf89e2e884e892940108de6ba48e94c3bcb5032ea5569ce708038f02ef67193d1448110f51b57e5d3132f0101e673ffcb96e9cdfcebb3b0e5fbc249708fff0104017e3594f904eed87647bb55c21b80308f1045ed28815f008b7345fcad044d2700083dfe30d2ef90c8e2c2b68747470733a2f2f626f622e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267fff01077c21f58df73633e11572092afee762808f1045ed28815f0085bf39858d2c3a9da0083dfe30d2ef90c8e2e2d68747470733a2f2f616c6963652e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267fff0108e27f6a15086a8e6d89b26094d2f99d208f1045ed28816f008fd133e3e5a683c480083dfe30d2ef90c8e292868747470733a2f2f66696e6e65792e63616c656e6461722e657465726e69747977616c6c2e636f6df010faeb530153e7239c6bea9db97855debe08f1045ed28816f008ebde0e542e11857f0083dfe30d2ef90c8e232268747470733a2f2f6274632e63616c656e6461722e636174616c6c6178792e636f6d)

</details>

File: test/functional/wallet_resendwallettransactions.py [hunk 14]
```diff
@@ -49,16 +49,21 @@ def run_test(self):
...
-        # Transaction should not be rebroadcast
         node.syncwithvalidationinterfacequeue()
-        node.p2ps[1].sync_with_ping()
-        assert_equal(node.p2ps[1].tx_invs_received[txid], 0)
+        now = int(time.time())
+
+        # Transaction should not be rebroadcast within first 12 hours
+        # Leave 2 mins for buffer
+        twelve_hrs = 12 * 60 * 60
+        two_min = 2 * 60
```
Comment (glowang) [inline test/functional/wallet_resendwallettransactions.py:58]: I'm curious - any reason why we chose 2 minutes here?

Comment (MarcoFalke) [reply test/functional/wallet_resendwallettransactions.py:58]: It can probably be reduced to one second, if `now` gets set to `block_time` (the mock time before this call)
